app/models/tag.rb

```diff
@@ -589,7 +599,7 @@ def self.find_or_create_by_name(new_name)
       elsif tag
         self.find_or_create_by_name(new_name + " - " + self.to_s)
       else
-        self.create(:name => new_name)
+        self.create(name: new_name, type: self.to_s)
```

Comment:
This is due to the bug we found in 4.0 with STI. If you run `Media.create(params)` it would result in a `Fandom` object getting created. If you explicitly add in the type param, it would maintain the proper type. Ideally, this could go away at 4.2 (ish) if the bug is resolved at that point.